crates/librqbit_core/src/magnet.rs

```diff
@@ -54,6 +60,32 @@ impl Magnet {
                     }
                 }
                 "tr" => trackers.push(value.into()),
+                "so" => {
+                    // Process 'so' values, but silently ignore any which fail parsing
+                    for file_desc in value.split(',') {
+                        if file_desc.is_empty() {
+                            continue;
+                        }
+                        // Handling ranges of file indices
+                        if let Some((start, end)) = file_desc.split_once('-') {
+                            let maybe_start_idx: Result<usize, _> = start.parse();
+                            let maybe_end_idx: Result<usize, _> = end.parse();
+                            if let (Ok(start_idx), Ok(end_idx)) = (maybe_start_idx, maybe_end_idx) {
+                                files.extend(start_idx..=end_idx);
+                            } else {
+                                warn!("magnet so validation failed with either invalid start or end index");
+                            }
+                        } else {
+                            // Handling single file index
+                            let idx = file_desc.parse();
+                            if let Ok(idx) = idx {
+                                files.push(idx);
+                            } else {
+                                warn!("magnet so validation failed with unknown file index {:?}", idx.err());
```

Comment:
for this and above: there'll be (or may be) too many warnings this way that aren't very helpful. E.g. "so=a,b,c,d" will produce 4 error lines.

let's either remove logging, or:
- downgrade to debug
- include url as tracing event
- for consistency, format the parameters as tracing events

i.e.

```
warn!(magnet=%url, err=?idx.err(), "magnet so validation failed: error parsing file index");
```

Reply:
dropped warnings altogether for now :) feel free to provide or push a commit with improved logging if you'd like :)